### ISSUE TITLE
Group Boards by Project/Client/Team

### ISSUE THREAD

**Issue by Quin452** [nextcloud/deck#2006]
Hi

Firstly, sorry if this has been addressed already, but I couldn't find any information about it (perhaps I am wording my searches wrong).

Basically, I was wondering if there was a way to group the boards in some way (like you can with Trello - these are called "Teams").
Essentially I have multiple projects on the go, constantly.  Some are my own websites, some are the games I'm working on, and some are with clients, or multiple projects with the same client.

It would be handy for me to quickly switch to my "game development boards" rather than have to scroll to find them.

**Comment by juliushaertl** [comment]
Do you have some example how that is implemented at trello in terms of setting the project and organizing boards in those?

**Comment by Quin452** [comment]
Hi @juliushaertl.  I would recommend having a look [here](https://help.trello.com/article/1177-personal-vs-team-boards), it probably gives the quickest and best explanation :-)

The Teams feature allows a lot more customisation, which I believe you've got implemented by integrating Circles (Board collaboration, etc.)

For a simple system for organisation (which is what I believe is missing/what I'd like), I'd recommend essentially adding a field to the Board, so it can be selected and filtered.
It could be free text, or an additional setting page where you can add groups to populate a dropdown/select box.

I'd honestly would offer to develop it myself, but I don't have the knowledge nor the time right now, and hoping it would be a quick addition.

**Comment by Dubidubiduu** [comment]
Hey @Quin452,

a simple and not so powerful workaround is to use the abc and generate prefixes, since it's ordered alphabetically. So g_ = games, w_ = websites or you order by client name...

**Comment by Quin452** [comment]
Hi @Dubidubiduu thanks for the suggestion.
The only problem I see with that is how can I group and view all the game boards, or website boards?

With Teams, and what I'm suggesting/would like, is the ability to group them together so I can view them all at once, rather than have to scroll through a huge list of boards, until I find "g_platformer" (because I have a lot of boards).

**Comment by krestyanikov** [comment]
Hello everyone.

I also really need this feature because the number of boards is very large. I'd suggest we see how it's done in MaesterTask

https://www.meistertask.com/videos/project_groups.mp4

Thanks!

**Comment by aproposnix** [comment]
Don't project groups already sort of exist? I mean you currently have two groups out of box "All Boards" and "archived boards". In this case, you would provide the user with a way of creating new ones.

**Comment by stefan-niedermann** [comment]
@aproposnix this is currently a flag:

`archived: true / false`

**If** grouping will be implemented, this mechanism will still have to work - each board in each group can be (or not be) `archived`. I doubt that this will be helpful for this feature.

**Comment by juliushaertl** [comment]
Yeah I see that this could be useful at some point so let's keep this open, however it is not on the near future roadmap.

**Comment by tomdereub** [comment]
I don't know if it would be useful or annoying for other people, but for me I would like also if boards in a same group could have the same parameters (number and name of columns, same labels, etc...). So that when I add a label it's added for all boards in the group.
For example, I would have a group "projects" with all projects I'm working on, and for all these projects I need the same columns and labels. For now, i have to add a new label manually on each board.

**Comment by stefan-niedermann** [comment]
@tomdereub you can clone an existing blard which will also clone the tags and lists (but not the card). So create a "template" board and clone it over and over again for your need.

It's an interesting idea to manage this stuff on a group level, but i fear it makes things even more complicated.

**Comment by juliushaertl** [comment]
Yeah, that might be a nice idea for later as well.

**Comment by Quin452** [comment]
> @tomdereub you can clone an existing blard which will also clone the tags and lists (but not the card). So create a "template" board and clone it over and over again for your need.
> 
> It's an interesting idea to manage this stuff on a group level, but i fear it makes things even more complicated.

Can the template board be used across projects?

**Comment by super-cooper** [comment]
I would love to see either this, or simply the ability to tag boards in the same way that we can tag cards.

**Comment by HonkXL** [comment]
This feature would be very useful for lot of users with many boards. My board list is really long and a kind of grouping / categorizing would make everything much easier.

**Comment by juliushaertl** [comment]
Makes sense to me.

**Comment by weeman1337** [comment]
Picking up the feedback from the PR #3325 I think it is still unclear how this should work.

These points have to be discussed and decided:
- How many "categories" should a board have?
  - max 1
  - unlimited (more like tagging, labeling, etc.)
- Should the "category" assignmed be user specific?
  - Store it per user and board
  - Store it per board (same for all users)
- How should the name of this feature be?
  - Category
  - Folder
  - Group
  - Tag

I really want to contribute the implementation work. But before these points should be decided. @juliushaertl @nextcloud/designers

**Comment by ilippert** [comment]
> These points have to be discussed and decided:
> 
>     * How many "categories" should a board have?
>       
>       * max 1
>       * unlimited (more like tagging, labeling, etc.)

I would very much prefer "unlimited", because boards, in my use cases, can be part of multiple work categories (e.g. a literature review board can be part of "category" teaching and research - in more detail at https://github.com/nextcloud/deck/pull/3325#issuecomment-974862274).

> 
>     * Should the "category" assignmed be user specific?
>       
>       * Store it per user and board
>       * Store it per board (same for all users)

I would argue it should be user specific insofar as, in my view, Nextcloud is very much about giving users _their_ own cloud experience.


@weeman1337, 
> "Shared with me" displays all boards shared with me (like it currently does)

yes, and no, I would say. That is, because: yes, of course the "category" of "Shared with me" should show the respective boards that are shared with me. But, no, because not only. I may want a board to be _also_ part of an own "category".

**Comment by HonkXL** [comment]
PR #3325 was a solution for this. OK, not perfect, but I think it would be a first step towards this feature. Why not using this "board-categories" as a kind of "board-default category". The creator of a board can set a default category.
Later, when it has programmed, the user can overwrite this for his own user with a user category and fallback to the board default category.

I think this would be better than just not having this feature.

We are heavy users of deck and if you really use it for projects, you are assigned to a lot of boards and it's not clear to find the right board. I first have to scroll through the list, this is sometimes confusing.

**Comment by ilippert** [comment]
I wonder, actually, whether this discussion is related to the older discussion at https://github.com/nextcloud/deck/issues/1280

**Comment by HonkXL** [comment]
We use a patched version that uses grouping for the board on a per board-level (not per user).
The feedback from our users is: we never want to miss this feature.

Here a idea how it looks here. I think it would be a great idea to see this in the official release.

![nc-category](https://user-images.githubusercontent.com/16715040/161228500-4717d141-b288-4a97-ba00-fba1264a5655.jpg)

**Comment by SchraderSarah** [comment]
The company I work for would like to have exactly the solution @HonkXL proposed.
We placed a [Bounty of 50$](https://app.bountysource.com/issues/91685078-board-grouping) on this issue.

**Comment by HonkXL** [comment]
It is programmed an running in our installation of Nextcloud. @weeman1337 did a good job implementing this for us. The only problem was, that we did not get this feature to the official repository. Now we have our really own Nextcloud with "own" features... Pushing this feature to the official repo would be possible, but our last try ended up in a long discussion about where this feature should appear and if it's to have a board grouping on board level instead of user level.
My experience from the last months in our company says: It's better to have this feature on a board level than not to have this feature....

**Comment by vanyatwo** [comment]
I would propose to implement it on the same behavior as Trello, Jira, etc. - just board level.
@HonkXL @weeman1337 could you please share a DIY solution to implement?

**Comment by compumatter** [comment]
I have moved #4041 to this location as it is clearly the same problem.

I have many boards that are sub-boards of a another board. Currently the 1 level linear approach requires all boards are viewed at once. This limits Decks reach to simple use cases. It would make more use of available board menu space and allow for easier comprehension of tasks to allow for a minimum of 2 levels but preferably many more.

The use of an jQuery collapsible or nestable accordion board list would do it best. They do not have to indent so no added width would be required. As one level is clicked only those things that are child projects would be displayed.

When adding the board as a new board, you would only need to ask in board details to [ Select Parent Board ] to make this board a child of the selected board.  If no parent is chosen, it would simply become a parent board and visible on page load.

I have created a video to provide a clear understanding what I think the solution would be.

- https://www.youtube.com/watch?v=wiP_YdbYyyw

I would add $300.00 to this bounty if this could be made a priority.  I notice this post is over 2 years old and I feel it's importance has been minimized.  For businesses like ours who will have perhaps 100 projects and many that are child projects, this is a required solution.   We cannot use Deck without such a solution.  Yet because Deck integrates with Nextcloud we would really like to see this happen.

**Comment by ianrandmckenzie** [comment]
Hi, new to the NextCloud community. Has a consensus been determined about how grouping should look/work? Not sure how to contribute to this discussion, but I'm available to keep this conversation going.

**Comment by Ornanovitch** [comment]
It sounds to me like the Collective way of organizing things could be a really nice inspiration. In a general sense, Nextcloud would really improve its UX by harmonizing the design and by harmonizing the way resources are organized across the featured collaboration apps (when its relevent: Collective, Deck, Tables).
